src/views/components/NetPromoterScoreForm/index.tsx

```diff
@@ -0,0 +1,272 @@
+        this.setState({ errors: "Please rate NSSF as a recommendation" });
+        return false;
+      }
+      if (!this.state.evaluation.crmCaptureCorrect) {
+        this.setState({ errors: "Please select if CRM capture" });
+        return false;
+      }
+      if (!this.props.frontLineReasons) {
+        this.setState({
+          errors: "Please select at least one Front line reason"
+        });
+        return false;
+      }
+      if (!this.props.backOfficeReason) {
+        this.setState({
+          errors: "Please select at least one Back Office reason"
+        });
+        return false;
+      }
+    }
+    this.setState({ errors: null });
+    return true;
+  };
+
+  public clearInput = () => {
+    const evaluation = {
+      ...this.state.evaluation,
+      firstTimeInteraction: "",
+      contact: 0,
+      date: "",
+      time: "",
+      touchPoint: "",
+      reason: "",
+      waitTime: "",
+      duration: "",
+      issueResolved: "",
+      furtherInformationGiven: "",
+      recommendationReason: "",
+      rateReason: "",
+      crmCaptureCorrect: "",
+      crmCaptureReason: "",
+      rangeValue: 0
+    };
+    this.setState({ evaluation });
+  };
+
+  public handleSubmit = async (): Promise<void> => {
+    if (this.validate()) {
+      this.setState({ loading: true });
+      this.props.onSubmit(this.state.evaluation);
+      this.clearInput();
+    }
+  };
+
+  public renderForm = () => {
+    if (this.state.target === "no") {
+      return (
+        <FirstTimeInteraction
+          evaluation={this.state.evaluation}
+          handleInput={this.handleInput}
+        />
+      );
+    } else if (this.state.target === "yes") {
+      return (
+        <Container>
+          <OnlineSection
+            evaluation={this.state.evaluation}
+            activeSection={this.state.activeSection}
+            agents={this.props.agents}
+            reason={this.props.reason}
+            crmTask={this.props.crmTask}
+            touchPoints={this.props.touchPoints}
+            handleInput={this.handleInput}
+            handleChangeRadioInput={this.handleChangeRadioInput}
+            handleSectionClick={this.handleSectionClick}
+            handleDropDownInput={this.handleDropDownInput}
+          />
+          <OfflineSection
+            activeSection={this.state.activeSection}
+            handleSectionClick={this.handleSectionClick}
+            frontLineRatingReason={this.props.frontLineReasons}
+            backOfficeReason={this.props.backOfficeReason}
+            handleDropDownInput={this.handleDropDownInput}
+          />
+        </Container>
+      );
+    }
+  };
+
+  public render() {
+    const hasError = !!this.state.errors;
+    return (
+      <Form
+        onSubmit={this.handleSubmit}
+        error={hasError}
+        loading={this.state.loading}
+      >
+        <Form.Field required={true}>
+          <label>
+            Good day, this is “….. Name” from NSSF, do you have 3 minutes to
```

Comment:
This would be logged in `Evaluator` so you can add an evaluator prop to the component
```suggestion
            Good day, this is {this.props.evaluator} from NSSF, do you have 3 minutes to
```

Reply:
@aleku399  you missed this!